up42/catalog.py

```diff
@@ -4,84 +4,179 @@
 
 import pathlib
 import warnings
-from typing import Any, Dict, List, Optional, Union
+from typing import Any, Dict, List, Literal, Optional, TypedDict, Union, cast
 
 import geojson  # type: ignore
 import geopandas  # type: ignore
 import tqdm
 from shapely import geometry as geom  # type: ignore
+from typing_extensions import TypeAlias
 
 from up42 import auth as up42_auth
-from up42 import host, order, utils
+from up42 import base, host, order, utils
 
 logger = utils.get_logger(__name__)
 
 
-class CatalogBase:
-    """
-    The base for Catalog and Tasking class, shared functionality.
-    """
-
-    def __init__(self, auth: up42_auth.Auth, workspace_id: str):
-        self.auth = auth
-        self.workspace_id = workspace_id
-        self.type: Optional[str] = None
+class ResolutionValue(TypedDict):
+    minimum: float
+    maximum: Optional[float]
+
+
+class Host(TypedDict):
+    # pylint: disable=invalid-name
+    name: str
+    title: str
+    description: str
+    createdAt: Optional[str]
+    updatedAt: Optional[str]
+    isIntegrated: bool
+    isCommercial: bool
+
+
+class Producer(TypedDict):
+    # pylint: disable=invalid-name
+    name: str
+    title: str
+    description: str
+    createdAt: Optional[str]
+    updatedAt: Optional[str]
+    isIntegrated: bool
+
+
+CollectionType: TypeAlias = Literal["ARCHIVE", "TASKING"]
+
+
+class Collection(TypedDict):
```

Comment:
I know this is something that we cant change because it comes from the endpoint. But we are adding an extra layer of confussing our users by naming collections in the catalog. We need to explain not only that STAC collection are assets but now that we have catalog collections that correspond to the providers datasets. In the ideal STAC world STAC Collection should correspond to Provider Collection.

Reply:
Perhaps some naming or documentation at the SDK level that we can help better the users to understand this.